### PR TITLE
Init/Bean context performance improvements

### PR DESCRIPTION
**Pull request by dstepanov**

### CODE REVIEW

File: inject/src/main/java/io/micronaut/context/env/DefaultEnvironment.java [hunk 1]
```diff
@@ -81,7 +82,7 @@
     private static final String ORACLE_CLOUD_ASSET_TAG_FILE = "/sys/devices/virtual/dmi/id/chassis_asset_tag";
     private static final String ORACLE_CLOUD_WINDOWS_ASSET_TAG_CMD = "wmic systemenclosure get smbiosassettag";
     private static final String DO_SYS_VENDOR_FILE = "/sys/devices/virtual/dmi/id/sys_vendor";
-    private static final Boolean DEDUCE_ENVIRONMENT_DEFAULT = true;
+    private static final Boolean DEDUCE_ENVIRONMENT_DEFAULT = GraphicsEnvironment.isHeadless();
```
Comment (dstepanov) [inline inject/src/main/java/io/micronaut/context/env/DefaultEnvironment.java]: This is up for discussion

Comment (graemerocher) [reply inject/src/main/java/io/micronaut/context/env/DefaultEnvironment.java]: I don't think we should depend on AWT

We can make a decision to default to `false` for 3.0 or at least default to not make the network calls for certain clouds

Comment (jameskleeh) [reply inject/src/main/java/io/micronaut/context/env/DefaultEnvironment.java]: Perhaps move this change and any other changes related to environment deduction to a separate PR so we can merge the rest of this?

Comment (graemerocher) [reply inject/src/main/java/io/micronaut/context/env/DefaultEnvironment.java]: Agreed

File: inject/src/main/java/io/micronaut/inject/annotation/AnnotationMetadataSupport.java [hunk 2]
```diff
@@ -94,7 +94,7 @@
      */
     @UsedByGeneratedCode
     public static Map<String, Object> getDefaultValues(String annotation) {
-        return ANNOTATION_DEFAULTS.computeIfAbsent(annotation, s -> Collections.emptyMap());
+        return ANNOTATION_DEFAULTS.getOrDefault(annotation, Collections.emptyMap());
```
Comment (dstepanov) [inline inject/src/main/java/io/micronaut/inject/annotation/AnnotationMetadataSupport.java:97]: This doesn't look like it was correct, inserting an immutable empty map where the value can be modified and some of the code checked if default values are present.

File: inject/src/main/java/io/micronaut/context/env/DefaultEnvironment.java (hunk 1 above)
Comment (dstepanov) [reply inject/src/main/java/io/micronaut/context/env/DefaultEnvironment.java]: I have removed it for now

**Pull request merged**